packages/manager/modules/pci/src/projects/project/public-gateways/edit/edit.controller.js

```diff
@@ -21,26 +24,43 @@ export default class PciPublicGatewaysEditController {
     this.selectedGatewaySize = selectedSize;
   }
 
+  getGatewayById() {
+    return this.PciPublicGatewaysService.getGatewayById(
+      this.projectId,
+      this.region,
+      this.gatewayId,
+    )
+      .then((data) => {
+        this.gatewayDetails = data;
+        this.gatewayName = this.gatewayDetails.name;
+        return this.gatewayDetails;
```

Comment:
Suggestion:  .then((gatewayDetails) => gatewayDetails)
.catch((err) => this.CucCloudMessage.error(err.data.message || err.data));